draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2611,39 +2777,137 @@ to the curve E' isogenous to BLS12-381 G1 is given in {{sswu-map-to-3mod4}}.
 
-The common parameters for the above suites are:
+BLS12381G2-XMD:SHA.256-SSWU-NU- is identical to BLS12381G2-XMD:SHA.256-SSWU-RO-,
+except that the encoding type is encode\_to\_curve ({{roadmap}}).
 
-- E: y^2 = x^3 + 4 * (1 + I)
-- p, m, F: defined above
-- sgn0: sgn0\_be ({{sgn0-be}})
-- H: SHA-256
-- L: 64
-- h\_eff: 0xbc69f08f2ee75b3584c6a0ea91b352888e2a8e9145ad7689986ff031508ffe1329c2f178731db956d82bf015d1212b02ec0ec69d7477c1ae954cbc06689f6a359894c0adebbf6b4e8020005aaa95551
+BLS12381G2-XMD:SHA.256-SVDW-NU- is identical to BLS12381G2-XMD:SHA.256-SVDW-RO-,
+except that the encoding type is encode\_to\_curve ({{roadmap}}).
 
-Note that this h\_eff value is chosen for compatibility
+Note that the h\_eff values for these suites are chosen for compatibility
 with the fast cofactor clearing method described by
 Budroni and Pintore ({{BP18}}, Section 4.1).
 
+## Defining a new hash-to-curve suite {#new-suite}
+
+The RECOMMENDED way to define a new hash-to-curve suite is:
+
+1. E, F, p, and m are determined by the elliptic curve and its base field;
+   k is determined by the security level of the elliptic curve.
+
+2. Choose encoding type, either hash\_to\_curve or encode\_to\_curve ({{roadmap}}).
+
+3. Choose a sgn0 variant following the guidelines in {{sgn0-variants}}.
+
+4. Compute L as described in {{hashtofield-sec}}.
+
+5. Choose an expand\_message variant from {{hashtofield-expand}} plus any
+   underlying cryptographic primitives (e.g., a hash function H).
+
+6. Choose a mapping following the guidelines in {{choosing-mapping}},
+   and select any required parameters for that mapping.
+
+7. Choose h\_eff to be either the cofactor of E or, if a fast cofactor
+   clearing method is to be used, a value appropriate to that method
+   as discussed in {{cofactor-clearing}}.
+
+8. Construct a Suite ID following the guidelines in {{suiteIDformat}}.
+
+When hashing to an elliptic curve not listed in this section, corresponding
+hash-to-curve suites SHOULD be specified as described in this section.
+
+## Suite ID naming conventions {#suiteIDformat}
+
+Suite IDs MUST be constructed as follows:
+
+    CURVE_ID || "-" || HASH_ID || "-" || MAP_ID || "-" || ENC_VAR || "-"
+
+The fields CURVE\_ID, HASH\_ID, MAP\_ID, and ENC\_VAR are
+ASCII-encoded strings of at most 64 characters each.
+Fields can contain only ASCII characters between 0x21 and 0x7E (inclusive)
+other than hyphen and underscore (i.e., 0x2d, and 0x5f).
+As indicated above, each field (including the last) is followed by a hyphen
+("-", ASCII 0x2d); this helps to ensure that Suite IDs are prefix free.
+
+Fields MUST be chosen as follows:
+
+- CURVE\_ID: a human-readable representation of the target elliptic curve.
+
+- HASH\_ID: a human-readable representation of the expand\_message function
+  and any underlying hash primitives used in hash\_to\_field ({{hashtofield}}).
+  This field MUST be constructed as follows:
+
+        EXP_TAG || ":" || HASH_NAME
```

Comment:
I think this is not needed, since a hash function cannot be used as an extendable output function and vice versa (unless there is an special case).
So, specifying the `expand_message` variant must be done inside the suite parameters, but not in the suiteID.

Reply:
The reason I was thinking to have EXP_TAG is that if someone introduces a new expand_message variant in the future, this keeps things unambiguous. I suppose it's true that we could just require them to tag their nonstandard variant, though. I'm pretty neutral on this. @chris-wood, thoughts?

Reply:
Thinking about it a little more, I think I do slightly prefer having an explicit tag. But only slightly...

Reply:
I too think being explicit is better, as it doesn't rely on properties of the expansion function for separation. (@armfazh how would you imagine someone define `HASH_ID` if, say, they introduced an expansion variant such as `expand_message_md_like`?)

Reply:
OK, since we now have a suite-ID issue, I'm going to assume that if we end up making a change to this we can do it there, too. I've added a link.